### PR TITLE
Introducing the Cache V1 task

### PR DESCRIPTION
**Pull request by owenhuynMSFT**

### CODE REVIEW

**Comment by Amoenus** [comment]
Just a thought. If that's a **Beta**, maybe consider assigning it a v0.1.0 version. 
Another option is to use (Preview) indicator

**Comment by owenhuynMSFT** [comment]
> Just a thought. If that's a **Beta**, maybe consider assigning it a v0.1.0 version.
> Another option is to use (Preview) indicator

We've introduced a breaking change in the task so it isn't as simple as switching the minor version. Since there's a change in the behavior, if we update just the minor version, everyone will get the changes since one cannot specify a specific minor version of the task. By switching major versions, one has to opt into the new behavior. @willsmythe can you confirm this?

File: Tasks/CacheBetaV1/task.json [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    "author": "Microsoft Corporation",
+    "version": {
+        "Major": 1,
+        "Minor": 0,
+        "Patch": 0
+    },
+    "groups": [],
+    "demands": [],
+    "preview": true,
+    "minimumAgentVersion": "2.155.0",
```
Comment (fadnavistanmay) [inline Tasks/CacheBetaV1/task.json]: 2.155.1

File: Tasks/CacheBetaV1/task.loc.json [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  "author": "Microsoft Corporation",
+  "version": {
+    "Major": 1,
+    "Minor": 0,
+    "Patch": 0
+  },
+  "groups": [],
+  "demands": [],
+  "preview": true,
+  "minimumAgentVersion": "2.155.0",
```
Comment (fadnavistanmay) [inline Tasks/CacheBetaV1/task.loc.json]: Same here - 2.155.1

File: Tasks/CacheBetaV1/task.json [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+            "type": "string",
+            "label": "Cache hit variable",
+            "defaultValue": "",
+            "required": false,
+            "helpMarkDown": "Variable to set to 'true' when the cache is restored (i.e. a cache hit), otherwise set to 'false'."
+        },
+        {
+            "name": "restoreKeys",
+            "type": "multiLine",
+            "label": "Additional restore key prefixes",
```
Comment (fadnavistanmay) [inline Tasks/CacheBetaV1/task.json]: IMO we should remove "additional", but I'll let @willsmythe take the call.

**Comment by willsmythe** [comment]
@Amoenus / @owenhuynMSFT - yah, to answer the earlier question: switching `key` from a multi-line string (newline-delimited) to a single-line string (pipe-delimited) is a breaking change and needs to be made in a major version of the task to avoid impacting existing pipelines.

**Pull request merged**